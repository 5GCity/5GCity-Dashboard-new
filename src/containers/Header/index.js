/**
 * Header Container
 * Please write a description
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React, { Component } from 'react'
import { compose } from 'recompose'
import { connect } from 'kea'
import Logic from './logic'
import styled from 'styled-components'

import Brand from 'components/Brand'
import Container from 'components/Container'

class Header extends Component {
  render () {
    return (
      <Wrapper>
        <SpacedContainer>
          <Brand />
        </SpacedContainer>
      </Wrapper>
    )
  }
}


export default compose(
  connect({
    props: [
      Logic, [

      ]
    ],
    actions: [
      Logic, [

      ]
    ]
  })
)(Header)

const Wrapper = styled.header`
  background-color: ${({ theme }) => theme.headerBackground};
`

const SpacedContainer = styled(Container)`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
`
