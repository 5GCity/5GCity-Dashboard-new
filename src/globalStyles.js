import { injectGlobal } from 'styled-components'

// Theme variables
export const Theme = {
  bodyBackground: '#37474F',
  headerBackground: '#eef1f6',
  primaryColor: '#8CC14E',
  secondaryColor: '#89979F',
  dangerColor: '#DD6C6C',
  fontFamily: 'Open Sans, sans-serif',
  fontDin: 'd-din, sans-serif',
  fontDinCondensed: 'd-din-condensed ,sans-serif',
  fontDinExp: 'd-din-exp ,sans-serif'
}

// Global styles
injectGlobal`
  @import url('https://fonts.googleapis.com/css?family=Open+Sans:400,600');

  body, html, #root , main {
    height: 100%;
  }

  body {
    margin: 0;
    padding: 0;
    font-family: ${Theme.fontFamily};
    text-rendering: optimizeLegibility;
    -webkit-font-smoothing: antialiased;
    text-shadow: 1px 1px 1px rgba(0, 0, 0, 0.004);
    -moz-osx-font-smoothing: grayscale;
    -ms-overflow-x: hidden;
    background: ${Theme.bodyBackground};

    .el-picker-panel {
      background: #404F57!important;
      border: none;
      color: white;
    }
    .el-date-table td.disabled {
      background-color: transparent;
      color: #6a6a6a;
    }
    .el-form-item__label {
      color: #fff;
    }
  }

  * {
    box-sizing: border-box;
  }

  /* width */
::-webkit-scrollbar {
  width: 5px;
}

/* Track */
::-webkit-scrollbar-track {
  background-color: rgba(0,0,0,0.1);
}

/* Handle */
::-webkit-scrollbar-thumb {
  background-color: #76848C;
  border-radius: 4px;
}

/* Handle on hover */
::-webkit-scrollbar-thumb:hover {
  background: #555;
}
/* MAPBOX */
.mapboxgl-marker > svg {
  position: absolute;
}
.mapboxgl-ctrl-logo {
  display: none !important;
}
`
