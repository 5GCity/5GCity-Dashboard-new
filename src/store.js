import { getStore } from 'kea'
import sagaPlugin from 'kea-saga'

export const Store = getStore({
  plugins: [ sagaPlugin ]
})
