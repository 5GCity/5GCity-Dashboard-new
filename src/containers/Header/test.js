/**
 * Header Container Tests
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React from 'react'
import renderer from 'react-test-renderer'
import Header from './index'
import 'jest-styled-components'

import { Store } from 'store'
import { Theme } from 'globalStyles'
import { ThemeProvider } from 'styled-components'
import { Provider } from 'react-redux'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={Theme}>
        <Provider store={Store}>
          <Header />
        </Provider>
      </ThemeProvider>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
