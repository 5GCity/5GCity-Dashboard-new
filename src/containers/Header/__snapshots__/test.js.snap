// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  background-color: #eef1f6;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  max-width: 1200px;
  margin: 0 auto;
}

.c3 {
  width: 100%;
  height: 88px;
  background-color: #FFFFFF;
  box-shadow: inset 0 -1px 0 0 rgba(137,151,159,0.15);
}

<header
  className="c0"
>
  <div
    className="c1 c2"
  >
    <div
      className="c3"
    />
  </div>
</header>
`;
